### PR TITLE
fix: Reuse the existing JitsiLocalTrack on presenter unmute

### PR DESCRIPTION
**Pull request by jallamsetty1**
- Reuse the existing JitsiLocalTrack for the presenter video on presenter unmute.
- Dispatch a new UIEvent PRESENTER_MUTED when the JitsiLocalTrack mute handler emits a TRACK_MUTE_CHANGED event after muting/unmuting the presenter track.
- handle device selection changes properly when screenshare is in progress.
- dispatch setVideoMuted only for camera/presenter track and not for desktop tracks.

### CODE REVIEW

File: react/features/base/tracks/middleware.js [hunk 1]
```diff
@@ -135,8 +135,13 @@ MiddlewareRegistry.register(store => next => action => {
             const isVideoTrack = jitsiTrack.type !== MEDIA_TYPE.AUDIO;
 
             if (isVideoTrack) {
-                if (jitsiTrack.isLocal()) {
-                    APP.conference.setVideoMuteStatus(muted);
+                if (jitsiTrack.type === MEDIA_TYPE.PRESENTER) {
+                    APP.UI.emitEvent(UIEvents.PRESENTER_MUTED, muted);
```
Comment (virtuacoplenny) [inline react/features/base/tracks/middleware.js]: What about calling `mutePresenter` directly instead of adding more to APP.UI, which is also legacy? Ideally not adding to conference.js would be nice but potentially not practical.

Comment (jallamsetty1) [reply react/features/base/tracks/middleware.js]: Done.

File: conference.js [hunk 2]
```diff
@@ -815,41 +815,34 @@ export default {
...
-        let effect;
-
-        try {
-            effect = await this._createPresenterStreamEffect(height,
-                defaultCamera);
-        } catch (err) {
-            logger.error('Failed to unmute Presenter Video');
-            maybeShowErrorDialog(err);
+            try {
+                effect = await this._createPresenterStreamEffect();
```
Comment (saghul) [inline conference.js]: These 2 blocks can be combined into one:

```
await this.localVideo.setEffect(await this._createPresenterStreamEffect());
```

File: conference.js [hunk 3]
```diff
@@ -1612,31 +1605,66 @@ export default {
      * @return {Promise<JitsiStreamPresenterEffect>} - A promise resolved with
      * {@link JitsiStreamPresenterEffect} if it succeeds.
      */
-    async _createPresenterStreamEffect(height, cameraDeviceId = null) {
-        let presenterTrack;
+    async _createPresenterStreamEffect(height = null, cameraDeviceId = null) {
+        if (!this.localPresenterVideo) {
+            let presenterTrack;
```
Comment (saghul) [inline conference.js]: why the auxiliary variable? you could assign it to `this.localPresenterVideo` directly

Comment (jallamsetty1) [reply conference.js]: Done

File: conference.js (hunk 2 above)
Comment (jallamsetty1) [reply conference.js]: Done

File: react/features/base/tracks/functions.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ export async function createLocalPresenterTrack(options, desktopHeight) {
         video: {
             aspectRatio: 4 / 3,
             height: {
-                exact: result
+                ideal: result
```
Comment (saghul) [inline react/features/base/tracks/functions.js]: out of curiosity, why this change?

Comment (jallamsetty1) [reply react/features/base/tracks/functions.js]: Sorry, should have mentioned the reason for the change. As per spec, when we specify exact, the property must have to be be considered to find a match while with ideal, it is considered optional. I am trying to avoid situations where GUM would fail because the camera doesn't support that particular resolution.

Comment (saghul) [reply react/features/base/tracks/functions.js]: Makes sense, thanks!

File: react/features/base/tracks/middleware.js [hunk 5]
```diff
@@ -135,7 +135,12 @@ MiddlewareRegistry.register(store => next => action => {
             const isVideoTrack = jitsiTrack.type !== MEDIA_TYPE.AUDIO;
 
             if (isVideoTrack) {
-                if (jitsiTrack.isLocal()) {
+                if (jitsiTrack.type === MEDIA_TYPE.PRESENTER) {
+                    APP.conference.mutePresenter(muted);
+                }
+
+                // Make sure we change the video mute state only for camera tracks.
+                if (jitsiTrack.isLocal() && jitsiTrack.videoType !== 'desktop') {
```
Comment (paweldomas) [inline react/features/base/tracks/middleware.js]: If the goal is to change only for camera tracks then maybe the condition should be if `videoType === 'camera'` ?
Also with the change now the logic will go to line 146 for local track which is meant for the remote participants. That seems incorrect.

Comment (jallamsetty1) [reply react/features/base/tracks/middleware.js]: I know this looks confusing but the here is what it is intended to do:
1. For the local camera tracks, regular video and presenter, we need to call UI.setVideoMuted() which updates the thumbnail view and notify external API of change in mute state (called inside conference.setVideoMuteStatus()).
2. For the remote tracks (camera+desktop) and for local desktop track, we need to update the thumbnail view so we need to call UI.setVideoMuted().
We don't support a secondary camera source now but if we were to add a new videoType for a secondary camera, then checking for `videoType !== 'desktop'` would still work. I could change it though :)

Comment (paweldomas) [reply react/features/base/tracks/middleware.js]: Ok, thanks for the explanation!
